### PR TITLE
Iot fix tests

### PR DESCRIPTION
**Pull request by gguuss**
Cleans up the old Java test registries when the tests run.

### CODE REVIEW

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 1]
```diff
@@ -173,6 +173,103 @@ public static void deleteRegistry(String cloudRegion, String projectId, String r
...
+            .execute()
+            .getDevices();
+
+    if (devices != null) {
+      System.out.println("Found " + devices.size() + " devices");
+      for (Device d : devices) {
+        String deviceId = d.getId();
+        String devicePath = String.format(
+            "%s/devices/%s", registryPath, deviceId);
+        service.projects().locations().registries().devices().delete(devicePath).execute();
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: Maybe the comment about remove should be here and the one above be different.

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 2]
```diff
@@ -173,6 +173,103 @@ public static void deleteRegistry(String cloudRegion, String projectId, String r
...
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    final CloudIot service =
+        new CloudIot.Builder(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init)
+            .setApplicationName(APP_NAME)
+            .build();
+    final String registryPath =
+        String.format(
+            "projects/%s/locations/%s/registries/%s", projectId, cloudRegion, registryName);
+
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: Pseudo code for what's going on?
```
ForAll gateways
  ForAll deviceNumIds
     UnbindDeviceFromGatewayRequest

ForAll Devices
  Delete DevicePath

DeleteRegistry
```

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 3]
```diff
@@ -173,6 +173,103 @@ public static void deleteRegistry(String cloudRegion, String projectId, String r
...
+    List<Device> gateways =
+        service
+            .projects()
+            .locations()
+            .registries()
+            .devices()
+            .list(registryPath)
+            .setGatewayListOptionsGatewayType("GATEWAY")
+            .execute()
+            .getDevices();
```
[the remark this answers is not in the document]
Comment (gguuss) [reply iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: Happy to break into separate statements.

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java (hunk 2 above)
Comment (gguuss) [reply iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: Yes, this is high-level what's going on.

**Pull request merged**